examples/intro_to_observations.jl

```diff
@@ -0,0 +1,145 @@
+# # Intro to observations
+#
+# This example demonstrates the features of `OneDimensionalTimeSeries`
+# when constructed from  "synthetic observations" generated by an Oceananigans `Simulation`.
+#
+# ## Install dependencies
+#
+# First let's make sure we have all required packages installed.
+
+# ```julia
+# using Pkg
+# pkg"add OceanTurbulenceParameterEstimation, Oceananigans, CairoMakie"
+# ```
+
+# First we load few things
+pushfirst!(LOAD_PATH, joinpath(@__DIR__, ".."))
+
+using OceanTurbulenceParameterEstimation
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.TurbulenceClosures: ConvectiveAdjustmentVerticalDiffusivity
+
+using CairoMakie
+# using ElectronDisplay #src
+
+# # Generating synthetic observations
+#
+# We define a utility function for constructing synthetic observations,
+
+function generate_free_convection_synthetic_observations(name = "convective_adjustment";
+                                                         Nz = 32,
+                                                         Lz = 64,
+                                                         Qᵇ = +1e-8,
+                                                         Qᵘ = -1e-5,
+                                                         Δt = 10.0,
+                                                         f₀ = 1e-4,
+                                                         N² = 1e-6)
+    data_path = name * ".jld2"
+
+    if isfile(data_path)
+        return data_path
+    end
+
+    convective_κz = 1.0
+    convective_νz = 0.9
+    background_κz = 1e-4
+    background_νz = 1e-5
+
+    grid = RegularRectilinearGrid(size=32, z=(-64, 0), topology=(Flat, Flat, Bounded))
+    closure = ConvectiveAdjustmentVerticalDiffusivity(; convective_κz, background_κz, convective_νz, background_νz)
+    u_bcs = FieldBoundaryConditions(top = FluxBoundaryCondition(Qᵘ))
+    b_bcs = FieldBoundaryConditions(top = FluxBoundaryCondition(Qᵇ), bottom = GradientBoundaryCondition(N²))
+
+    model = HydrostaticFreeSurfaceModel(grid = grid,
+                                        tracers = :b,
+                                        buoyancy = BuoyancyTracer(),
+                                        boundary_conditions = (; u=u_bcs, b=b_bcs),
+                                        coriolis = FPlane(f=f₀),
+                                        closure = closure)
+                                        
+    set!(model, b = (x, y, z) -> N² * z)
+    
+    simulation = Simulation(model; Δt, stop_time=12hours)
+
+    init_with_parameters(file, model) = file["parameters"] = (; Qᵇ, Qᵘ, Δt)
+    
+    simulation.output_writers[:fields] = JLD2OutputWriter(model, merge(model.velocities, model.tracers),
+                                                          schedule = TimeInterval(4hour),
+                                                          prefix = name,
+                                                          array_type = Array{Float64},
```

Comment:
I feel we should remark somewhere about `array_type = Array{Float64}`?

Reply:
We can. It only matters for _testing_ though, not for normal runs. Maybe we should remove it from the example, in fact... ?